src/Analyzers.ruleset

```diff
@@ -4,18 +4,20 @@
          ToolsVersion="14.0">
 
   <Rules AnalyzerId="Microsoft.ApiDesignGuidelines.Analyzers" RuleNamespace="Microsoft.ApiDesignGuidelines.Analyzers">
-    <!-- DoNotDecreaseInheritedMemberVisibility - blocked by https://github.com/dotnet/roslyn-analyzers/issues/940 -->
-    <Rule Id="CA2222" Action="None" />
-  </Rules>
-
-  <Rules AnalyzerId="Microsoft.ApiDesignGuidelines.Analyzers" RuleNamespace="Microsoft.ApiDesignGuidelines.Analyzers">
     <!-- MarkAssembliesWithAssemblyVersion - suppressed because assembly versions for our analyzers are stored in https://github.com/dotnet/roslyn-analyzers/blob/master/build/Targets/Analyzers.Versions.targets -->
     <Rule Id="CA1016" Action="None" />
-  </Rules>
-
-  <Rules AnalyzerId="Microsoft.ApiDesignGuidelines.Analyzers" RuleNamespace="Microsoft.ApiDesignGuidelines.Analyzers">
     <!-- IdentifiersShouldNotContainTypeNames - not valuable as we don't plan to ship or support our public API surface. -->
     <Rule Id="CA1720" Action="None" />
+    <!-- DoNotDecreaseInheritedMemberVisibility - blocked by https://github.com/dotnet/roslyn-analyzers/issues/940 -->
+    <Rule Id="CA2222" Action="None" />
+    <!-- IdentifiersShouldNotContainUnderscores - test names contain underscores -->
+    <!-- Fix tracked by https://github.com/dotnet/roslyn-analyzers/issues/1354 -->
+    <Rule Id="CA1707" Action="None" />
```

Comment:
Instead of turning this off for all the analyzer projects, can we just have this suppression in test project specific ruleset?

Reply:
Seems like we got did of separate rulesets for analyzer and test projects. Can you replicate the structure we have in Roslyn repo where we have Roslyn_BuildRules.ruleset for shipping projects and NonShippingProject_BuildRules.ruleset for the test projects (https://github.com/dotnet/roslyn/blob/master/build/Rulesets)?

Reply:
We set the correct ruleset to use over here: https://github.com/dotnet/roslyn/blob/master/build/Targets/Imports.targets#L96